File: h/src/api/modules/systemConfig.ts
import request from '@/api/request'

export type ConfigCategory = 'PARAM' | 'INTERFACE' | 'PUSH' | 'STORAGE' | 'BACKUP'

export interface SystemConfig {
  id?: number
  key: string
  value?: string
  category: ConfigCategory
  updatedAt?: string
}

export const getCategories = () => {
  return request.get<string[]>('/api/v1/system/configs/categories')
}

export const listByCategory = (category: ConfigCategory) => {
  return request.get<SystemConfig[]>(`/api/v1/system/configs/${category}`)
}

export const setConfig = (category: ConfigCategory, key: string, value: string) => {
  return request.put<SystemConfig>(`/api/v1/system/configs/${category}/${key}`, { value })
}

File: campus-card-admin-api/src/main/java/com/campus/card/admin/domain/SystemConfig.java
package com.campus.card.admin.domain;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@Entity
@Table(name = "system_config")
public class SystemConfig {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "cfg_key", nullable = false, length = 64, unique = true)
    private String key;

    @Column(name = "cfg_value", length = 512)
    private String value;

    @Column(name = "category", length = 32)
    private String category; // e.g. PARAM, INTERFACE, PUSH, STORAGE, BACKUP

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}